pkg/reader/generic/sql.go

```diff
@@ -42,27 +42,45 @@ func (s *SqlReader) ReadTable(tableName string, rowChan chan<- database.Row, opt
 	logger := log.WithField("table", tableName)
 	logger.Info("Fetching rows")
 
-	columns, err := s.GetColumns(tableName)
+	if len(opts.Columns) == 0 {
```

Comment:
Should this be a func on the `ReadTableOpt`? something like `opt.ReadColumns(rdr)`

Reply:
Why? I don't think we will change this implementation for each call to `ReadTable`

Reply:
Just to make the code a little more readable but you can leave it the way it is

Reply:
Yes, I see. I wouldn't encapsulate this in the option parameters since this function will need the `GetColumns` from the reader. I would encapsulate in the reader. We can do this later if it gets ugly.